### DISCUSSION TITLE
Same Program runs gives different output

### DISCUSSION THREAD

**Discussion by JamalShams** [cyberbotics/webots#2739] [Q&A]
I made a PID controller in python language. It was built successfully, and it was taking off (almost) perfectly. in the first run it takes off perfect, but when I  run it again, it does not take off. what is this problem? I mean, the code is the same not a single change was made then why it is not running the same again?

**Comment by omichel** [comment]
Did you check this [list](https://cyberbotics.com/doc/guide/modeling#how-to-make-replicabledeterministic-simulations)?
